java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessConnection.java

```diff
@@ -247,10 +249,8 @@ public void setReadOnly(boolean readOnly) throws SQLException {
                 Constants.SQLExceptionMessages.METHOD_CALLED_ON_OPEN_TRANSACTION);
         }
 
-        if (readOnly) {
-            throw new SQLFeatureNotSupportedException(Constants.SQLExceptionMessages.READ_ONLY);
-        }
-        this.readOnly = false;
+        // TODO implement read only properly
+        this.readOnly = readOnly;
```

Comment:
Code that I am not in control over makes a optimization for MySQL and sets the read only flag.

We can fairly easily enforce it by not allowing `executeUpdate` to be called.